iree/compiler/Dialect/HAL/Target/LLVM/LLVMAOTTarget.cpp

```diff
@@ -224,8 +229,14 @@ class LLVMAOTTargetBackend final : public TargetBackend {
       llvmFunc->setDSOLocal(true);
       libraryBuilder.addEntryPoint(entryPointOp.getName(), "", llvmFunc);
     }
-    auto *queryLibraryFunc =
-        libraryBuilder.build("iree_hal_executable_library_query");
+
+    auto queryFunctionName = std::string(kQueryFunctionName);
+    if (!options_.staticLibraryOutput.empty()) {
+      // Static library query functions must be unique to support multiple
+      // libraries in the same namespace.
+      queryFunctionName = libraryName + "_library_query";
```

Comment:
I thought about that. Since different configs are activated by flags and its already a bit tricky to validate flag state before acting upon, I figured it was best to stay simple and (using the library name) for now. The users can see the output from the generated header (which those have to be unique anyway) so they know what the query is called.

Reply:
Sounds good. Since I want to put more thought into the flags, I'll do this as a followup. 

One thing, can you clarify on:
> names must be unique and library names need not be

Wouldn't the library names need to be unique as well since each library is referenced by name in the static loader?